socket_transport.go

```diff
@@ -93,6 +95,13 @@ func (st *socketTransport) writer() {
 				continue
 			}
 
+			lastBeat := time.UnixMilli(st.lastHeartbeat.Load())
+
+			if time.Now().Sub(lastBeat) > pongWait && !st.getIsReconnecting() {
+				st.reconnect <- struct{}{}
```

Comment:
What's this line doing? Something with channels?The empty struct literal is confusing me.

Reply:
Ah I think it's just something to trigger this bit of the `supervisor` method
```go
select
...
case <-st.reconnect:
```